client/blocks/reader-join-conversation/dialog.jsx

```diff
@@ -10,20 +10,35 @@ import './style.scss';
 const ReaderJoinConversationDialog = ( { onClose, isVisible, loggedInAction, onLoginSuccess } ) => {
 	const translate = useTranslate();
 
-	const { login, createAccount } = useLoginWindow( { onLoginSuccess: onLoginSuccess } );
-
 	const trackEvent = ( eventName ) => {
 		let eventProps = {};
 		if ( loggedInAction ) {
-			eventProps = {
-				type: loggedInAction?.type,
-				blog_id: loggedInAction?.siteId,
-				post_id: loggedInAction?.postId,
-			};
+			eventProps = { type: loggedInAction?.type };
+			// create eventProps from loggedInAction properties if they exist
+			if ( loggedInAction?.siteId ) {
+				eventProps.blog_id = loggedInAction?.siteId;
+			}
+			if ( loggedInAction?.postId ) {
+				eventProps.post_id = loggedInAction?.postId;
+			}
+			if ( loggedInAction?.tag ) {
+				eventProps.tag = loggedInAction?.tag;
+			}
```

Comment:
Curious why the format change here? I think they should have the same effect in the end. IIRC if one of the values in the red object definition are null the track event doesn't add those properties anyways.

Reply:
I saw they were being passed as `undefined` so figured better to only pass them if set.

Reply:
I see that `recordTracksEvent` removes any undefined properties so good to leave this as it was.